Add padding for months

danamansana opened the pull request:
Fixes a small bug where months less than 10 are not being padded correctly.

yossariano reviewed (dismissed):
Approved- but if we keep playing whack-a-mole with date issues I'd be in favor of us splitting the date and subject reindexing so one is not blocking the other.

nonword reviewed (dismissed):
Given that Sierra is applying subtly different validation to this data than we are, is it inevitable that we'll encounter a value that has been validated by Sierra, and properly reformatted by us, but rejected by ES - e.g. `2025-04-31`? We either need to replicate that validation here.. Or place a final catch around the update call to retry saving the document with nullified dates. Replicating ES date parsing in JS seems tough:
```
> new Date('2025-04-31')
2025-05-01T00:00:00.000Z
```
🙃

danamansana commented:
@nonword I've added `moment` to try to validate dates as much as possible before attempting to write to the index. Can't guarantee that it will match ES validation exactly, however testing with ~20k records so far nothing unexpected. We could try to add some logic to retry records without dates in case of errors related to dates, but this seems complicated to me, let me know if you have an idea of how to do that.

nonword commented on lib/es-models/bib.js:
```diff
@@ -296,6 +297,10 @@ class EsBib extends EsBase {
       const day = second.slice(2, 4)
       dates.push(generateDateRangeFromSingleDate(first, month, day, rawMarc, type))
```
Should all invocations of `generateDateRangeFromSingleDate` be wrapped in a try-catch? There may be date type conditions where we happen to know that's not likely, but the caller shouldn't know that.

danamansana commented on lib/utils/es-ranges.js:
```diff
@@ -76,6 +77,18 @@ const roundDateUp = (date) => {
 }
 
 class InconsistentDateRange extends Error {}
+class InvalidDate {}
```
It seemed like passing a value rather than an error would be a nice way to get the same functionality with less error handling. Given your comment below I can rewrite it as an error if that seems preferable

danamansana commented:
@nonword  I've rewritten the code you commented on to use `InvalidDateError` , I think this addresses all the points you raise, let me know if not. Also added a test and fixed some merge issues (although not merging yet)
